src/schema-builder/SchemaBuilder.ts

```diff
@@ -95,19 +95,23 @@ export class SchemaBuilder {
     // Private Methods
     // -------------------------------------------------------------------------
 
+    protected get entityToSyncMetadatas(): EntityMetadataCollection {
+        return <EntityMetadataCollection>this.entityMetadatas.filter( metadata => !metadata.table.skipSchemaSync );
```

Comment:
also your code is not correct. `.filter` does not return instance `EntityMetadataCollection`, it returns you `Array`.  So you are lying to compiler by casting it to `EntityMetadataCollection` will bring you errors once there will be a call to a specific method.